### ISSUE TITLE
Issue: Unable to use dependency-cruiser when using babel-macro in a project

### ISSUE THREAD

**Issue by mesteche** [sverweij/dependency-cruiser#410]
### Expected Behavior

Dependency cruiser should run without issues when babel macros are used

### Current Behavior

When you have babel-macro configured in your project, the following error occur in files where you're using some macros:

```
  ERROR: Extracting dependencies ran afoul of...

  unknown: paths.macro: The "path" argument must be of type string. Received undefined Learn more: https://www.npmjs.com/package/paths.macro
... in src/index.js
```

### Possible Solution

This is probably due to the way babel is ran, it looks like babel is missing some information.

### Steps to Reproduce (for bugs)

After cloning the minimal example repo : https://github.com/mesteche/depcruise-babel-macro-bug, run npm install and try runing depcruise: 
`npx depcruise --config .dependency-cruiser.js src`

note that the babel config is fine and webpack is able to bundle the project without any issue, only depcruise fails to run.

### Context

In the example I used the paths-macro but I'm also having this bug with styled-components macro in a react project.

### Your Environment

- Version used: ^9.21.1
- Node version: v15.5.0
- Operating System and version: linux
- Link to your project: https://github.com/mesteche/depcruise-babel-macro-bug

**Comment by sverweij** [comment]
Hi @mesteche thanks for this bug report - and for the laser focused example repro!

The root cause of the problem is that the `paths.macro` module assumes a piece of source code has a filename (in [macro.js#L50](https://github.com/storybookjs/paths.macro/blob/master/macro.js#L50), which code runs by way of a side-effect of require-ing it). In case there is no filename, `babel-plugin-macros` doesn't pass one to it either. It seems `babel-plugin-macros` _is) aware this can happen judging  [this piece of code](https://github.com/kentcdodds/babel-plugin-macros/blob/master/src/index.js#L192)).

The best fix would be that paths.macros and/ or babel-plugin-macros apply checks on their inputs. However, the easiest way  is to have dependency-cruiser pass a filename. PR #411 does that.

I've published `dependency-cruiser@9.21.2-beta-1` which includes the fix. I've run it against the minimal example repo and it seems to work.

=> Could you check on your side whether it fixes the issue for you too (maybe also against the real repo where you bumped into this?)

**Comment by mesteche** [comment]
Hello again @sverweij,
I just tested your fix and I can confirm it works with `styled-components/macro`, as well as `paths.macro`.
These are the only two macros I am using for now but it seems you nailed it.

Thanks for the reactivity, it works like a charm.

**Comment by sverweij** [comment]
hi @mesteche thanks for checking so quickly! I'll merge the PR and release the fix as part of the regular _latest_ release somewhere in the coming days.

**Comment by sverweij** [comment]
(published as part of `dependency-cruiser@9.21.2`)
